src/qibolab/sweeper.py

```diff
@@ -85,17 +85,18 @@ class Sweeper:
     values: npt.NDArray
     pulses: Optional[list] = None
     qubits: Optional[list] = None
+    couplers: Optional[list] = None
     type: Optional[SweeperType] = SweeperType.ABSOLUTE
 
     def __post_init__(self):
         if self.pulses is not None and self.qubits is not None:
             raise ValueError("Cannot use a sweeper on both pulses and qubits.")
         if self.pulses is not None and self.parameter in QubitParameter:
-            raise ValueError(f"Cannot sweep {self.parameter} without specifying qubits.")
-        if self.qubits is not None and self.parameter not in QubitParameter:
+            raise ValueError(f"Cannot sweep {self.parameter} without specifying qubits or couplers.")
+        if self.parameter not in QubitParameter and self.qubits is not None or self.couplers is not None:
```

Comment:
Here some parenthesis might help, but at the moment you're not able to create a `Sweeper` with `self.couplers` in any case (`or` has lower [precedence](https://docs.python.org/3/reference/expressions.html#operator-precedence) than `and`, and it's even coming later, so if it is not `None` you're always entering the `if`, and consequently raising an error).

If this is desired, I do not understand why `self.couplers` appears even in the following.
If this is not, we're missing a test for the sweeper with couplers.

Reply:
I would like to have a `Sweeper` with couplers. I will try to add a test and have it working.

Reply:
Here the `Sweeper` has the whole `Coupler` or `Qubit`. I think we can remove couplers from the `Sweeper` and just use qubits in this case.

Or change that line to:

```py
 if (self.parameter not in QubitParameter and self.qubits is not None) or (self.parameter not in QubitParameter and self.couplers is not None):
            raise ValueError(f"Cannot sweep {self.parameter} without specifying pulses.")
```